watcher.go

```diff
@@ -0,0 +1,67 @@
+package consul
+
+import (
+	"context"
+	"strconv"
+)
+
+type WatcherFunc func([]KeyData)
+
+type Watcher struct {
+	Client *Client
+}
+
+var (
+	DefaultWatcher = &Watcher{Client: DefaultClient}
```

Comment:
Can we follow the pattern used in other types where the zero-value is valid and a `nil` Client indicates that the default client should be used?